agent/engine/task_manager.go

```diff
@@ -415,8 +415,15 @@ func (task *managedTask) progressContainers() {
 	task.UpdateStatus()
 }
 
-func (task *managedTask) cleanupTask() {
-	cleanupTime := ttime.After(task.KnownStatusTime.Add(taskStoppedDuration).Sub(ttime.Now()))
+func (task *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
+	cleanupTimeDuration := task.KnownStatusTime.Add(taskStoppedDuration).Sub(ttime.Now())
+	// There is a potential deadlock here if cleanupTime is negative. Ignore the computed
+	// value in this case in favor of the default config value.
```

Comment:
Using the minimum clearly matches the user intention where the default clearly does not. This would have messed me up a lot if I didn't review these changes. With my original code I was using `1s` seemingly without issue. This code would have put me at `3h`, crashing my instances again, whereas `1m` clearly matches my intention the best and would work just fine.